w3-blob.md

```diff
@@ -380,15 +426,23 @@ type BlobPut = {
 
 ### Put Blob Subject
 
-The subject field SHOULD be [`did:key`] corresponding to the [Ed25519] private key that is first 32 bytes of the blob [multihash].
+The subject field SHOULD be [`did:key`] corresponding to the [Ed25519] private key that is last 32 bytes of the blob [multihash].
 
 ### Put Blob Metadata
 
-Metadata MUST contain `keys` field with an object value that contains [`did:key`] subject as key and corresponding private key as bytes as a value.
+Metadata MUST contain `keys` field with an object value that contains [`did:key`] subject as key and corresponding private key bytes as a value.
+
+### Put Blob URL
+
+Destination `url` MUST be specified in the arguments and it MUST be an endpoint that can accept HTTP PUT request.
+
+### Put Blob Headers
+
+The `headers` map MUST be specified in the arguments. It MUST have string keys corresponding to header names and string values corresponding to header values.
 
-### Put Blob Address
+### Put Blob Body
```

Comment:
Per @alanshaw https://github.com/web3-storage/w3up/pull/1342/files#r1557547248

> Personally I'd just repeat the fields here so this doesn't depend on blob capability.


I think it is a valid point, and I have considered defining schema as following instead

```ts
type BlobPut = {
  cmd: "/http/put"
  sub: DID
  args: {
    url: URL
    headers: Headers
    body: Body
  }
  meta: {
    keys: {[key: DID]: bytes}
  }
}

type Body = Variant<{
  blob: Blob
}>
```

So we could in the future define alternative body types. However I'm not sure if there could be a body that we do not address by hash which is why decided to not introduce another wrapper layer, that said I could be convinced otherwise.

Also relatedly I'm hoping to change `Blob` definition into a multihash that captures payload size per https://github.com/multiformats/multihash/issues/163, which is why `Blob` is treated as a single a unit as opposed to pair of fields.

Reply:
One other thing we have discussed is changing body to just multihash and moving `size` into `headers['content-length']` which we can do if that is preferred. From what I know I don't think presigned URLs currently care about `size` anyway.

Reply:
I would really prefer to have the size instead of expecting the size to be in the headers.